app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -258,15 +243,15 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
     }
 
     fun dispatchKeyEvent(event: KeyEvent): Boolean {
-        /**
-         * Key handling order:
-         * - Menu to control overlay
-         * - Youtube remap of BACK to ESC
-         * - Cursor
-         * - Return false, as unhandled
-         */
+        val handledCursorEvent = serviceLocator?.cursorModel?.handleKeyEvent(event)
+
+        handledCursorEvent?.simulatedTouch?.let {
+            activity?.dispatchTouchEvent(it)
+            it.recycle()
+        }
+
         return handleSpecialKeyEvent(event) ||
-                (cursor?.keyDispatcher?.dispatchKeyEvent(event) ?: false)
+                handledCursorEvent?.wasConsumed == true
```

Comment:
My understanding of this key dispatch pattern is that if one function returns true, that function handled the key press and subsequent functions should not get to handle it. However, the way this is written, `handleSpecialKeyEvent` and `cursorModel.handleKeyEvent` could both handle a key event.

I'd like to keep this pattern. I think it's helpful because it allows the developer to consider a smaller scope: "if this module (i.e. function) handles this key event, I do not need to consider any others" rather than having to check every function for whether it handles a particular key (here and back in MainActivity).

Maybe a reasonable phrasing that minimally changes behavior is:
```kotlin
fun handleCursorKeyEvent(event: KeyEvent): Boolean {
    val handledCursorEvent = serviceLocator?.cursorModel?.handleKeyEvent(event)
    handledCursorEvent?.simulatedTouch?.use {
        activity?.dispatchTouchEvent(it)
    }
    return wasConsumed == true
}

return handleSpecialKeyEvent(event) ||
    handleCursorKeyEvent(event)